src/main/java/ui/listpanel/ListPanel.java

```diff
@@ -564,11 +564,12 @@ private void markAsRead() {
         }
     }
 
-    private void createNewWatchListMenuItem() {
+    /**
+     * Creates a new watch list panel containing the currently selected issue
+     */
+    private void createNewWatchListPanel() {
         Optional<GuiElement> item = listView.getSelectedItem();
-        if (!item.isPresent()) {
-            return;
-        }
+        assert item.isPresent();
```

Comment:
A bit late, but I'm not sure the item can disappear after the right click and before the menu appears. Concurrency is not an issue here, as UI operations all run on the JavaFX thread, i.e. if the items in a list can swapped out before the menu appears, it must be happening by design, which seems unlikely.

The optional there captures the possibility of there not being anything selected. In this case, we've shown the context menu, which is triggered by selection, so there must be a selected item. An assertion seems like the right thing to do then.